freeipa/client/init.sls

```diff
@@ -6,6 +6,77 @@ include:
 - freeipa.client.nsupdate
 - freeipa.client.cert
 
+{%- if pillar.freeipa.client.install_principal is defined %}
+{% set otp = salt['random.get_str'](20) %}
+
+push_principal:
+  file.managed:
+    - name: /tmp/principal.keytab
+    - source: {{ client.get("install_principal", {}).get("source", "salt://freeipa/files/principal.keytab") }}
+    - mode: {{ client.get("install_principal", {}).get("mode", 0600) }}
+    - user: {{ client.get("install_principal", {}).get("file_user", "root") }}
+    - group: {{ client.get("install_principal", {}).get("file_group", "root") }}
+    - unless:
+      - ipa-client-install 2>&1 | grep "IPA client is already configured on this system"
+get_ticket:
+  cmd.run:
+    - name: kinit {{ client.get("install_principal", {}).get("principal_user", "root") }}@{{ client.get("realm", "") }} -kt /tmp/salt-service.keytab
```

Comment:
I'm not sure I understand - are you suggesting that this be done as kinit && examplecommand && kdestroy? As it is now, the principal only pushes out if IPA Client install does not report "IPA client is already configured on this system" in which case, it is assumed that a host needs to be created on the IPA server for which a kerberos ticket is required, so kinit is only called when needed and destroyed after we are done.